File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 7/25/2026, 7:29:54 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap;

        for (int i = 0; i < nums.size(); i++) {
            if (hashMap.find(target - nums[i]) != hashMap.end()){
                return {i, hashMap[target - nums[i]]};
            }
            hashMap[nums[i]] = i;
        }

        return {};
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 7/25/2026, 7:29:52 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        for i, num in enumerate(nums):
            if (target - num) in hashMap:
                return [i, hashMap[target - num]]
            hashMap[num] = i
        return []
        